### PR TITLE
p4, jean duran villanueva

### PR DESCRIPTION
**Pull request by duranueva**

### CODE REVIEW

File: DuranVillanuevaJean/practica3/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# PREGUNTAS -------------------------
+
+## ¿Qué diferencias encuentran entre trabajar con textos en español y en Otomí?
+
+Que la estructura del idioma es bastante distinta a lo que había notado con otros idiomas más parecidos: italiano, español, ingles.
+
+Además, que supuso todo un reto manejar estos datos por mi corta experiencia, pues el modelo no podía codificar los datos sin un tratamiento previo.
```
Comment (umoqnier) [inline DuranVillanuevaJean/practica3/README.md:7]: Esto suele pasar con lenguas que no son inglés O: hay herramientas asumen muchas cosas de la lengua :p

File: DuranVillanuevaJean/practica3/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+## ¿Se obtuvieron buenos o malos resultados? ¿Porqué?
+
+Creo que los resultados fueron favorables si los comparo con los de mis compañeras que obtuvieron resultados similares, aun cuando ellas me comentan que agregaron más atributos a los datos de entrenamiento (en X en particular). En general, creo además que tuve resultados aceptables, sobre todo con Accuracy (que refleja que tan bien, donde 0 es el peor resultado y 1 es el mejor, se predijo correctamente a los datos). Tuve flaqueza con lás demás metricas (precision, recall, f1-score), aunque se mantuvieron constantes entre 0.73 y 0.75.
```
Comment (umoqnier) [inline DuranVillanuevaJean/practica3/README.md:11]: Conviene no valerse solo del `accuracy` ya que puede ser engañoso. Si bien se obtiene un alto accuracy conviene mirar también las otras métricas para contrastar. La métrica que puede ser más informativa es el f1-score :D

File: DuranVillanuevaJean/practica3/corpus_otomi
Comment (umoqnier) [inline DuranVillanuevaJean/practica3/corpus_otomi:1]: No era necesario subir el corpus :open_mouth:

File: DuranVillanuevaJean/practica3/p3.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+  import ast
+  encoding = 'utf-8'
+
+  with open(path, 'r',encoding=encoding) as file:
+    # Iterate over each line in the file
+    for line in file:
+      try:
+        raw_corpus.append(eval(line)) # Use eval() to convert the line into a Python list
+        #print(raw_corpus)
```
Comment (umoqnier) [inline DuranVillanuevaJean/practica3/p3.py:38]: Estos coments podrías quitarlos para dejar un código más limpio

File: DuranVillanuevaJean/practica3/p3.py [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+  def word_to_features(sent, i):
+      word = sent[i][0]
+      features = {
+          'word.lower()': word.lower(),
+          'word[:3]': word[:3],
+          'word[:2]': word[:2],
+          'word.isupper()': word.isupper(),
+          'word.istitle()': word.istitle(),
+          'word.isdigit()': word.isdigit(),
```
Comment (umoqnier) [inline DuranVillanuevaJean/practica3/p3.py:163]: Estas features parecen ser inútiles, ya que el corpus está en minúsculas y no hay números

File: DuranVillanuevaJean/practica3/p3.py [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+      #print([word,tag])
+      lst.append([word,tag])
+    return lst
+
+  """
+  Take the parts of a word  and contats them
+
+  [['n', 'psd'], ['dó', '1.cpl'], ['phu̱di', 'stem']] -> ["ndóphu̱di"]
+  """
+  def aux(raw_word):
```
Comment (umoqnier) [inline DuranVillanuevaJean/practica3/p3.py:96]: Este nombre podría mejorar a otro más descriptivo, ¿no?

File: DuranVillanuevaJean/practica3/p3.py [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+
+  X = [[word_to_features(sent, i) for i in range(len(sent))] for sent in corpus]
+
+
+  """## Entrenando modelo"""
+
+  # Split the data into training and testing sets
+  X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+
+  from inspect import Attribute
```
Comment (umoqnier) [inline DuranVillanuevaJean/practica3/p3.py:193]: Esto no se ocupa para nada. Quitar O:

File: DuranVillanuevaJean/practica3/p3.py [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+  """
+  print("####################\nRecall")
+  from sklearn.metrics import recall_score
+  print(recall_score(y_pred_flat, y_test_flat, average="macro"))
+
+
+  """#### F1-score = $\frac{2PR}{P + R} = 0.7300134993972263$"""
+  print("####################\nF1-score")
+  from sklearn.metrics import f1_score
+  print(f1_score(y_pred_flat, y_test_flat, average="macro"))
```
Comment (umoqnier) [inline DuranVillanuevaJean/practica3/p3.py:249]: Conviene indicar que estás reportando la precision, recall y f1-score macro y no otra ya que es una tarea multi-class :)

**Review by umoqnier — APPROVED** [review]
Funcional, aunque veo que varias funciones y variables tienen nombre cuestionables y poco claros. Poner más esfuerzo en ese sentido ya que es importante.

Por otro lado, dejo unos coments en el análisis. Me parece valioso lo que notas sobre los problemas de codificación que surgieron con la herramienta y la lengua :)

Calificación: 9 + extra
